cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -181,6 +181,23 @@ ProgramDescriptor loadProgram(ProgramId program) throws IOException, Application
   Collection<ApplicationSpecification> getAllApplications(NamespaceId id);
 
   /**
+   * Returns a collection of all application specs of all the versions of the application by id
+   *
+   * @param id application id
+   * @return collection of all application specs of all the application versions
+   */
+  Collection<ApplicationSpecification> getAllAppVersions(ApplicationId id);
+
+  /**
+   * Returns a list of all versions' ApplicationId's of the application by id
+   *
+   * @param namespace application's namespace
+   * @param application returned string of getApplication() in ApplicationId
+   * @return collection of versionIds of the application's versions
+   */
+  Collection<ApplicationId> getAllAppVersionsAppIds(final String namespace, final String application);
```

Comment:
It is called in a scenario with only namespace and application, so thought it would be more convenient like this. Change it as suggested for consistency.